networking-calico/networking_calico/agent/dhcp_agent.py

```diff
@@ -502,9 +519,10 @@ def _pre_snapshot_hook(self):
         # information.
         LOG.debug("Reset cache for new snapshot")
         for network_id in list(self.agent.cache.get_network_ids()):
+            network = self.agent.cache.get_network_by_id(network_id)
             self.dirty_networks.add(network_id)
             self._fix_network_cache_port_lookup(network_id)
-            self.agent.cache.put(empty_network(network_id))
+            self.agent.cache.put(empty_network(network_id, network.mtu))
```

Comment:
I think we don't need these changes, because it's OK for networks to have the default MTU at this point?  As endpoints are processed from the snapshot, the relevant networks' MTU will be updated as needed.

Reply:
Yes, good point, had also overthought this.

Reply:
Removed in latest patch, still tests good.